packages/piro/src/Piro_NOXSolver_Def.hpp

```diff
@@ -161,12 +162,14 @@ void Piro::NOXSolver<Scalar>::evalModelImpl(
               std::ostringstream ss; ss << "Parameter Vector Index " << i;
               const int p_ind = rolParams.get(ss.str(), i);
               const auto paramNames = *this->getModel().get_p_names(p_ind);
-              this->observer->parameterChanged(paramNames[0]);
+              if(Teuchos::nonnull(this->observer))
+                this->observer->parameterChanged(paramNames[0]);
             }
             optimizationParams.template set<bool>("Optimization Variables Changed", false);
```

Comment:
You shouldn't need `template` here if `optimizationParams` is just a `Teuchos::ParameterList`, because its type does not depend upon a template parameter, and the type of the argument is known (`bool`).  Also, you don't need to say `bool` here; it's inferred from the input argument `false`.

Reply:
Thanks, I'll fix this. I remeber that I had to use `template` in a getter to avoid some mysterious build error on some machine. However here I don't need the bool, as you say.